### DISCUSSION TITLE
Disable UI or enable for LAN only

### DISCUSSION THREAD

**Discussion by jpergentino** [wg-easy/wg-easy#644] [Q&A]
Thank you for this amazing solution. It's really easy to configure and use it.

Is there a way to disable the WebUI or create a restriction to access via LAN only (or via an already connected client)?

Thank you in advance.

**Comment by zw135113** [comment]
I think it can be achieved through iptables rules.

**Comment by zw135113** [comment]
To restrict access to the local machine's TCP port 51821 so that only the 10.0.0.0/24 IP range can access it, you can use the following commands:
First, allow the 10.0.0.0/24 IP range to access port 51821:
`iptables -A INPUT -p tcp --dport 51821 -s 10.0.0.0/24 -j ACCEPT`

Next, deny access to that port from any other IP address:
`iptables -A INPUT -p tcp --dport 51821 -j DROP`

**Reply by jpergentino** [reply]
Thank you for your suggestion. I tried including this in the ufw (Ubuntu) and it didn't work. I will try again later and let you know.

Assuming my local LAN is 192.168.1.x and the VPN IP range is 10.8.0.x, do you think the range below will work?
192.168.0.0/24 and 10.0.0.0/24 ?

Thanks in advance.

**Reply by zw135113** [reply]
If it's a ufw rule, I think it should be configured like this:

Allow specific IP access

`ufw allow from 10.8.0.0/24 to any port 51821 proto tcp`

`ufw allow from 192.168.1.0/24 to any port 51821 proto tcp`

Deny access from other IPs

`ufw deny proto tcp to any port 51821`

I hope this can help you, my friend.

**Reply by jpergentino** [reply]
No, it didn't work because it seems the Docker iptables rules (opened ports when using -p param) executed when the docker is starting overrides the ufw rules.

I am afraid there is no other way to do that :-(

I am running this command:

`docker run -d --name=wireguard -e WG_HOST=<PRIVATE> -e PASSWORD=<PRIVATE> -e WG_DEFAULT_DNS="8.8.8.8, 8.8.4.4" -e WG_ALLOWED_IPS=192.168.1.0/24  -v ~/.wg-easy:/etc/wireguard -p 51820:51820/udp -p 51821:51821/tcp --cap-add=NET_ADMIN --cap-add=SYS_MODULE --sysctl="net.ipv4.conf.all.src_valid_mark=1" --sysctl="net.ipv4.ip_forward=1" --restart unless-stopped ghcr.io/wg-easy/wg-easy`
